### PR TITLE
Chart fixes and improvements 

### PR DESCRIPTION
**Pull request by stefanprodan**
* Fix SSH key mapping
* Restart operator on helm repositories changes
* Add description fields to CRD template
* Add liveness and readiness probes to chart and deploy yaml
* Add `HelmRelease` example to chart notes
* Bump helm chart version

### CODE REVIEW

File: deploy/flux-helm-release-crd.yaml [hunk 1]
```diff
@@ -39,57 +39,41 @@ spec:
           required: ['chart']
           properties:
             releaseName:
-              description: The Helm release name. If not supplied, it will be generated by affixing the
```
Comment (hiddeco) [inline deploy/flux-helm-release-crd.yaml]: Why are these removed?

Comment (stefanprodan) [reply deploy/flux-helm-release-crd.yaml]: Looks like make generate-deploy removes them, I have to add the description in the crd template.

Comment (stefanprodan) [reply deploy/flux-helm-release-crd.yaml]: I wander why the CI didn't find this in #24

**Review by hiddeco — APPROVED** [review]
